library/src/device/generator/generator.kernel.hpp

```diff
@@ -342,21 +347,25 @@ namespace StockhamGenerator
             /*=======================================================*/
             std::string loop;
             loop += "\tif(dim == 1){\n";
-            loop += "\t\t" + offset_name + " += counter_mod*" + stride_name + "[1];\n";
+            loop += "\t\t" + offset_name1 + " += counter_mod*" + stride_name1 + "[1];\n";
+            if (output == true) loop += "\t\t" + offset_name2 + " += counter_mod*" + stride_name2 + "[1];\n";
```

Comment:
for the generated two lines of code. I do not see the "+" sign.   They should be correct

```
		iOffset += counter_mod*stride_in[1];
		oOffset += counter_mod*stride_out[1];
```